docs/guides/src/main/server/caching.adoc

```diff
@@ -10,52 +10,110 @@ includedOptions="cache cache-*">
+* *keys*
+
+Local caches for realms, users, and authorization are configured to have 10,000 entries per default.
+The local key cache can hold up to 1,000 entries per default and defaults to expire every one hour.
+Therefore, keys are forced to be periodically downloaded from external clients or identity providers.
 
-* realms
-* users
-* authorization
-* keys
+In order to achieve an optimal runtime and avoid additional round-trips to the database you should consider looking at
+the configuration for each cache to make sure the maximum number of entries is aligned with the size of your database. More entries
+you can cache, less often the server needs to fetch data from the database. You should evaluate the trade-offs between memory utilization and performance.
 
 .Invalidation of local caches
 Local caching improves performance, but adds a challenge in multi-node setups.
+
 When one Keycloak node updates data in the shared database, all other nodes need to be aware of it, so they invalidate that data from their caches.
-The `work` cache is used for sending these invalidation messages.
+
+The `work` cache is a replicated cache and used for sending these invalidation messages. The entries/messages in this cache are very short-lived,
+and you should not expect this cache growing in size over time.
 
 .Authentication sessions
-Authentication sessions are started when an unauthenticated user or service tries to log in to Keycloak.
-The `authenticationSessions` distributed cache is used to save data during authentication of a particular user.
+Authentication sessions are created whenever a user tries to authenticate. They are automatically destroyed once the authentication process
+completes or due to reaching their expiration time.
+
+The `authenticationSessions` distributed cache is used to store authentication sessions and any other data associated with it
+during the authentication process.
+
+By relying on a distributable cache, authentication sessions are available to any node in the cluster so that users can be redirected
+to any node without loosing their authentication state. However, production-ready deployments should always consider session affinity and favor redirecting users
+to the node where their sessions were initially created. By doing that, you are going to avoid unnecessary state transfer between nodes and improve
+CPU, memory, and network utilization.
 
 .User sessions
-The following are the distributed caches for sessions of authenticated users and services:
+
+Once the user is authenticated, a user session is created. The user session tracks your active users and their state so that they can seamlessly
+authenticate to any application without being asked for their credentials again. For each application the user authenticates with a client session
+is created too so that the server can track the applications the user is authenticated with and their state on a per-application basis.
```

Comment:
```suggestion
is created too, so that the server can track the applications the user is authenticated with and their state on a per-application basis.
```